collect_app/src/main/java/org/odk/collect/android/utilities/MultiClickGuard.java

```diff
@@ -0,0 +1,27 @@
+package org.odk.collect.android.utilities;
+
+import android.os.SystemClock;
+
+public class MultiClickGuard {
+    private static final int CLICK_DEBOUNCE_MS = 1000;
+
+    private static long lastClickTime;
+    private static String lastClickName;
+
+    private MultiClickGuard() {
+    }
+
+    // Debounce multiple clicks within the same screen
+    public static boolean allowClick(String className) {
+        long elapsedRealtime = SystemClock.elapsedRealtime();
+        boolean isSameClass = className.equals(lastClickName);
+        boolean isBeyondThreshold = elapsedRealtime - lastClickTime > CLICK_DEBOUNCE_MS;
+        boolean isBeyondTestThreshold = lastClickTime == 0 || lastClickTime == elapsedRealtime; // just for tests
```

Comment:
I'm not sure I understand this. Does this assume no time passes between clicks in tests? As far as I can see this is `true` if not time has passed since the last click. In both espresso and normal junit it's very possible for a millisecond to pass between clicks. Maybe I just need some more caffeine ☕️.

Reply:
Yes, someone has added it on purpose so I guess it fixed the problem he had run into.

Reply:
Generally it shouldn't be a problem in which test a situation like that could happen that you need to click on the same element twice quickly? We keep `lastClickName` so for example if you click a button which opens an activity go back and click again it will work even if i took less than 1s.

Reply:
Is doesn't need to be the same element though right? It just needs to be in the same class (generally we're passing the Activity). Personlly the solution confuses me and I'm sure there are simpler ways to work out if we're in tests or not. I can look at later seeing as I have a personal vendetta against the line 😁. Let's just make sure to run `connectedAndroidTest` to check Espresso doesn't run into any problems with all the new guards.

Reply:
> Is doesn't need to be the same element though right?

right.